models/settings_helpers.go

```diff
@@ -186,6 +193,13 @@ func ValidateSettings(params *ChangeSettingsParams) error {
 		}
 	}
 
+	if params.DisableSTT && params.EnableSTT {
+		return fmt.Errorf("enable STT and disable STT cannot be both true")
+	}
+	if params.EnableSTT && !params.EnableTelemetry {
+		return fmt.Errorf("cannot enable STT while telemetry is disabled")
```

Comment:
You are confusing the current state and desired state there. We should not allow user to enable STT and disable cemetery in the same request (so `!params.EnableTelemetry` should be `params.DisableTelemetry`).

And then in other place, where we save, we should check that STT can't be enabled when telemetry is disabled.